### PR TITLE
Unit tests for websockets

### PR DESCRIPTION
**Pull request by sandlis**
Adding tests for websockets which would help make sure any newly added middlewares pass these test

### CODE REVIEW

File: api/api_test.go [hunk 1]
```diff
@@ -0,0 +1,147 @@
...
+	"net/http/httptest"
+	"net/http/httputil"
+	"net/url"
+	"strconv"
+	"strings"
+	"testing"
+
+	"github.com/raintank/tsdb-gw/api/models"
+	"github.com/raintank/tsdb-gw/auth"
+	"golang.org/x/net/http/httpguts"
```
Comment (Dieterbe) [inline api/api_test.go]: interesting package..

File: api/api_test.go [hunk 2]
```diff
@@ -0,0 +1,147 @@
...
+		if !bs.Scan() {
+			t.Errorf("backend failed to read line from client: %v", bs.Err())
+			return
+		}
+		fmt.Fprintf(c, "backend got %q\n", bs.Text())
+	}))
+
+}
+
+func TestReverseProxyWebSocketWithHijacker(t *testing.T) {
```
Comment (Dieterbe) [inline api/api_test.go]: as a reader, this is hard to follow. some comments would be helpful.
what are we actually testing for here? why are we adding routes here and not testing existing ones? what is the X-Scope-OrgID header?

**Comment by Dieterbe** [comment]
looks pretty good, just a bit hard to follow for me. see comments.

File: api/api_test.go (hunk 2 above)
Comment (sandlis) [reply api/api_test.go]: @Dieterbe Added comments and removed X-Scope-OrgID since its not required.
Please check whether it looks good.
I am adding new route since different commands have different routes, wanted to make it a generic test.
